### DISCUSSION TITLE
Use without MQTT and WiFi?

### DISCUSSION THREAD

**Discussion by EdwardSoaresJr** [HASwitchPlate/openHASP#256] [General]
Is it possible to use this without MQTT and WiFi? I am replacing the Mazda Commander in my 2015 Mazda 3 with a WT32-SC01 or similar. I want to avoid mosquitto, nodered and especially WiFi on my pi carputer if I can. I can use I2C, or SERIAL or? Instead of MQTT and WiFi. I can program whatever is needed on my end most likely but OpenHASP is kind of beyond my skill currently. I will be using it to provide media buttons with a local rotary encoder for volume/mute. Extended functionality to control relays and garage door hopefully. 
I am already building out an arduino pro mini to connect almost everything control wise and plugging it into the pi as a keyboard to control Open Auto Pro. My DIY capacitive touch buttons for the LCD screen, the rotary encoder for volume, the steering wheel buttons etc. It is more of a collector of all controls and then fed into the pi with one connection and I would like to feed the output of my WT32-SC01 to this also.
I was thinking I can use the serial debugging with proper parsing for this? MQTT dirties it quite a bit though without a connection. 

The WT32-SC01 has serial TX/RX btw :) at the debug lights that work otherwise the headers only have TX :( So this could work if I could figure out how to disable MQTT locally. Hoping there is already something built in someone can let me know about. I would have to run a second pi to use this with my use case as I need to give the output as a keyboard command and running mqtt and node red on the car pi will not allow this without some crazy programming sending it back out to the arduino anyway. I would like to keep the pi as offloaded as possible anyhow.

I would like to use this as it works on the WT32-SC01 very nicely, I am struggling to use LVGL directly on this and just build a static GUI against a communications library hence why I am trying to find a workaround for this and will eventually be using this in my house on HA anyway once I can figure out how to make it work over zigbee if possible instead of WiFi due to everything security being on UPS and batteries except wifi. Large house with multiple PoE Access Points. So I would like to learn it more anyway.

**Comment by fvanroie** [comment]
> Is it possible to use this without MQTT and WiFi?

Not out if the box, no. openHASP is concieved around MQTT as it is the mostly used and versatile Home Automation protocol.
You could disable the MQTT client entirely and try compiling with `HASP_USE_MQTT=0` but no guaranties...

All commands are setup to pass through a dispatcher. This sends the input/output along to the other components, like MQTT.
It should be feasible to add another protocol there, so that the input and output is structured in a consistent way.

The debug output is just for troubleshooting. No effort has been made to make the output machine readable.

Learning LVGL or maybe using 8ms.xyz might be a better fit for your project, but I haven't used the latter myself.
There are mictopython builds with LVGL for ESP32 as well if you are used to coding in Python.
Soon there will be the release of EdgeLine as an LVGL GUI builder... There are many ways to go about building a UI.

**Reply by EdwardSoaresJr** [reply]
Appreciate the input, 8ms is absolute trash IMO. EdgeLine still feels a bit too raw, my struggle is getting the drivers for LCD and Touch working on any other software for this device such as LVGL, still got a lot to learn. I will play around with disabling MQTT and and adding another com to the dispatcher while trying to learn embedded stuff more. I do have some python experience but mostly come from desktop, web and electronics background focused on PHP. The ESP, Arduino and pi etc blows my mind. I appreciate your hard work on this software and will be buying coffees soon. Spent a little too much on tinker hardware at the moment.

**Reply by nagyrobi** [reply]
@EdwardSoaresJr I think this would be a great addition as it would allow usage in different scenarios than we're used in home automation.

**Reply by bensuffolk** [reply]
I came here to ask this same question. I want to disable WiFi (can be after I have it configured) and then just use the serial port. It would be talking to a master arduino based board to send commands and receive button presses etc.

Did you make any progress?

**Reply by fvanroie** [reply]
I've added the option to compile the most recent `0.6.3-dev` firmware with these compile flags:
```ini
   -D HASP_USE_WIFI=0
   -D HASP_USE_MQTT=0
   -D HASP_USE_MDNS=0
   -D HASP_USE_OTA=0
   -D HASP_USE_HTTP=0
   -D HASP_USE_TELNET=0
   -D HASP_USE_FTP=0
```
Beware that you're basically stripping all network modules from openHASP and creating a custom firmware. This is not a supported build, but you can try it if you want/need it... You will have to rely on `config/*` commands to configure the device. There is no option to edit or upload files over serial.

You can configure the flash filesystem using the official build and then re-flash with the custom build when all the files need are uploaded. Alternatively you need to use a tool like `mklittlefs` to create and flash the data partition yourself.

**Reply by bensuffolk** [reply]
Thanks,

I'm not opposed to having the network available to switch on via serial to use web services etc for file uploads / configuration. Just want to be able to switch it off again after, and not use MQTT at all.

If I could find a nice looking device (like the Lanbon L8) with wired ethernet and PoE I'd be very happy indeed to use the network. I just can't use WiFi in my scenario.

I will check out the different options you have provided.

Once again thank you, amazing project!

**Reply by fvanroie** [reply]
Some network services like `http`, `telnet` and `ftp` can be started and stopped on demand, but not the others.
Using openHASP *without* any network stack hasn't been on my radar so it will require some logic to handle that...

I'm not really up to that challenge as there are more pressing issues at the moment, so the best next thing I can offer is using these compile flags to omit the modules altogether.

**Reply by bensuffolk** [reply]
For the moment I’m using with WiFi. And putting a dedicated WiFi point as part of the setup (I am still looking at non WiFi options, ideally if I could hack a wired Ethernet point onto a Lanbon L8 I’d be very happy)

**Reply by fvanroie** [reply]
This got me thinking... The W5500 needs 4 GPIOs for SPI (possibly 5 with IRQ) so one pre-made POE board comes to mind:

You might be able to power the face-plate only with a [PoE-FeatherWing](https://hackaday.io/project/168356-poe-featherwing) and use `K1`, `K2`, `K3` and `IO0` pins for Ethernet SPI.
I haven't traced `BP`, `HUMI` or `CF` from the pin-header to the board. Maybe some of those are also usable as GPIO.

If need be, `RX` and `TX` can be repurposed as IRQ, GPIO or even I2C for reading the MAC-chip (which is not necessary since the ESP32 already has a MAC address).

Depending if you have the US or EU version it will be easier/harder to fit it into a wallbox... I can imagine a *converter PCB* with angled pin-header on the front for the Lanbon face-plate and dupont Header on the back to plug the POE-Featherwing. Add some mounting holes and it could be a best-of-both-worlds solution!

**Reply by bensuffolk** [reply]
I normally use the USR-ES1 in my projects to keep the form factor right down, I have only used it with the ATMega processors though not the ESP. But have always managed to just used the 4 lines for SPI. I like the idea of POE though, so that one you linked to would be good. I generally use a separate POE module that plugs in line and gives 5V out, but they take up extra space as well.

I was contemplating designing a new daughter board to replace the power supply / relay board that comes with the L8. Its the EU (Im in the UK) version that I have to work with, which means space is tight.

**Comment by matsekberg** [comment]
A nice solution would be if mqtt in and out could be replaced with serial in and out.
Same commands and same responses.

**Comment by TeodorPresident** [comment]
Thanks . Its interesting !
